### PR TITLE
[MIRROR] Syringe gun changes, atomized

### PR DESCRIPTION
**Pull request by AustationBot**
Original PR: https://github.com/BeeStation/BeeStation-Hornet/pull/2399
--------------------
## About The Pull Request
-Piercing syringes now inject through thick clothing when used normally, instead of only being useful with a syringe gun
-Syringes shot from a syringe gun, or used from harm intent with a delay, now embed, inject their contents over time.  The initial injection is 5 units, and 1 unit each tick after that. It takes a little while to pull a syringe out, and, if failed, injects another five units

## Why It's Good For The Game

chemcombat is not very fun for anyone but the user, and most darts are a delayed one-shot kill. this is designed to give the victim a fighting chance by lowering reagents injectable per-hit from the more common methods of injection

## Changelog
:cl:
tweak: piercing syringes now inject past thick clothing when not shot from a syringe gun as well
balance: syringe guns no longer instantly inject their reagents, instead injecting them over time
add: syringes can be embedded on harm intent, also injecting reagents over time, but at a slower rate
fix: explosive chem mixes no longer purge chems if an explosion is not triggered
/:cl:

### CODE REVIEW

**Comment by Kalvinalvaro** [comment]
zesk momento?

**Comment by thelaughingbomb** [comment]
seems like a buff to some degree since you can harm intent to embed syringes

**Comment by coolsurf6** [comment]
Wow
That is the worse fucking nerf I've ever seen.

**Comment by coolsurf6** [comment]
If you have chems that must react together instantly, and at full units, then this breaks it. You can't just shoot people with a tatp or explosive cryo syringe now, as it would just use the first 5 units, and barely do shit, then just keep fucking making mini explosions every tick.

**Comment by thelaughingbomb** [comment]
more knockdowns!

**Comment by TheFakeElon** [comment]
Good thing cryo explosive syringes were unbalanced one click wins

**Comment by MrLostman** [comment]
Hmmm seems like something that looks good from the top down but bad from the perspective of people who play the game. 

How many people have you seen actually use 1 hit explosive syringes?

Now think about how many people use things like sleepy pens, zombie powder, chloral hydrate and morphine this is a huge buff to these and thats not including toxins which it is also a huge buff too. Making all syringes sleepy pens, and lets you sleepy pen through armour unless you cant harm intent with a piercing syringe.

**Comment by MrLostman** [comment]
Oh smooth brained forgot to mention this **hugely exacerbates** the problem of being 1 tile next to someone is the most dangerous thing you can do. Making it basicly mandatory to never let anyone come 1 tile around you.

**Comment by TheFakeElon** [comment]
this isn’t as subtle as a sleepy pen, it injects over time rather than the instant however-many-units injection that comes from the sleepy pen. Most things besides zombie powder and.... _sarin_ shouldn’t be too hard to deal with.

HOWEVER, there is a less noticeable buff. now you can stab people with used syringes or syringes with infected blood instantly, this makes tot viro a little bit better here as target viruses with low infective rates can be stabbed into people.

Probably not worth the effort though so I doubt we’ll see it often

**Comment by TheFakeElon** [comment]
Would be nice if it had a short action timer when stabbing

**Comment by MrLostman** [comment]
While its not a 1 to 1 sleepy pen. It injects 5u instantly thats tones, then they have to remove it or get the full load so if you open with this in a fight you win unless they have something that can instantly deal with you

**Comment by MrLostman** [comment]
Like just think about what this adds
>no more explosive syringes
>widely available melee chem combat
melees chem combat will make a whole you meta

**Comment by MCterra10** [comment]
why
time to stall until the conflicts overwhelm me

**Comment by TheFakeElon** [comment]
if epi pens contained a small amount of penetic, I think it'd balance it out

**Comment by TheFakeElon** [comment]
chem combat is just as cancer right now, I think this is an overall improvement. But it could be improved further

**Comment by MrLostman** [comment]
how is it an overall improvement? 
You could get what this offers by removing cryo syringes without having melee combat be dictated by who stabs the other person first with a syringe.

In the first place who even complains about explosive syringes I don't think I haven even seen anyone use them.

Removing something that almost no one uses to replace it with something caner that everyone will use.

**Comment by TheFakeElon** [comment]
Hmmmm fair

**Comment by zeskorion** [comment]
> If you have chems that must react together instantly, and at full units, then this breaks it. You can't just shoot people with a tatp or explosive cryo syringe now, as it would just use the first 5 units, and barely do shit, then just keep fucking making mini explosions every tick.

...yes. this is a good change.

**Comment by TheFakeElon** [comment]
That part is yeah. But people running around jamming syringes into people isn’t. Now your opponent needs to make a choice between dying to chems or dying to a toolbox while removing said syringe. And that’s assuming the syringe doesn’t instantly incapacitate them or give them a lethal dose with the intitial injection.

Also virologists running around with aids syringes :(

**Comment by MCterra10** [comment]
yeah, the viro thing is actually fucking cringe. this is what I bugfixed on bee a while ago, where the entire station was getting infected because of instant syringe virus transfers.

**Comment by TheFakeElon** [comment]
Surprise surprise zesko buffing his main job again 😏

**Comment by TheFakeElon** [comment]
Bad bot 🗞️

**Comment by thelaughingbomb** [comment]
how long are we going to be putting this off for?

**Comment by jupyterkat** [comment]
long

**Comment by thelaughingbomb** [comment]
should just merge it smh

**Comment by ZombiesVsPlants** [comment]
god never please

**Comment by TheFakeElon** [comment]
fun fact: using new syringes would allow people to use syringe guns (with empty syringes) as effective ballistic weapons due to the damage of each projectile + embed

**Comment by ZombiesVsPlants** [comment]
that's actually hilarious

**Comment by zeskorion** [comment]
> fun fact: using new syringes would allow people to use syringe guns (with empty syringes) as effective ballistic weapons due to the damage of each projectile + embed

no u dumas it's not real embedding its a special thing im not a fuckn idiot

**Comment by MrLostman** [comment]
doubt

**Comment by TheFakeElon** [comment]
> no u dumas it's not real embedding its a special thing im not a fuckn idiot

I've heard otherwise

**Comment by TheFakeElon** [comment]
looking at the code it appears you're right actually. I'll ask around on bee later

File: code/modules/projectiles/projectile/bullets/dart_syringe.dm [hunk 1]
```diff
@@ -14,15 +15,21 @@
 		if(blocked != 100) // not completely blocked
 			if(M.can_inject(null, FALSE, def_zone, piercing)) // Pass the hit zone to see if it can inject by whether it hit the head or the body.
 				..()
-				reagents.reaction(M, INJECT)
-				reagents.trans_to(M, reagents.total_volume)
-				return BULLET_ACT_HIT
+				if(syringe)
+					syringe.embed(M)
```
Comment (TheFakeElon) [inline code/modules/projectiles/projectile/bullets/dart_syringe.dm:19]: Lol never mind. It does embed @zeskorion

**Comment by zeskorion** [comment]
The embed proc is a apecial prov used by syringes. Itit's not actual embedding

**Comment by MCterra10** [comment]
honestly, I haven't seen any issues while this was testmerged. full merging.

**Pull request merged**